matchzoo/utils/get_file.py

```diff
@@ -0,0 +1,425 @@
+    and given the filename `fname`. The final location of a file
+    `example.txt` would therefore be `~/.matchzoo/datasets/data/example.txt`.
+
+    Files in tar, tar.gz, tar.bz, and zip formats can also be extracted.
+    Passing a hash will verify the file after download. The command line
+    programs `shasum` and `sha256sum` can compute the hash.
+
+    :param fname: Name of the file. If an absolute path `/path/to/file.txt` is
+        specified the file will be saved at that location.
+    :param origin: Original URL of the file.
+    :param untar: Deprecated in favor of 'extract'. Boolean, whether the file
+        should be decompressed.
+    :param md5_hash: Deprecated in favor of 'file_hash'. md5 hash of the file
+        for verification
+    :param file_hash: The expected hash string of the file after download.
+        The sha256 and md5 hash algorithms are both supported.
+    :param cache_subdir: Subdirectory under the cache dir where the file is
+        saved. If an absolute path `/path/to/folder` is specified the file
+        will be saved at that location.
+    :param hash_algorithm: Select the hash algorithm to verify the file.
+        options are 'md5', 'sha256', and 'auto'. The default 'auto' detects
+        the hash algorithm in use.
+    :papram extract: True tries extracting the file as an Archive, like tar
+        or zip.
+    :param archive_format: Archive format to try for extracting the file.
+        Options are 'auto', 'tar', 'zip', and None.
+        'tar' includes tar, tar.gz, and tar.bz files.
+        The default 'auto' is ['tar', 'zip'].
+        None or an empty list will return no matches found.
+    :param cache_dir: Location to store cached files, when None it defaults to
+        the [matchzoo.USER_DATA_DIR](~/.matchzoo/datasets).
+
+    :return: Path to the downloaded file.
+    """
+    if md5_hash is not None and file_hash is None:
+        file_hash = md5_hash
+        hash_algorithm = 'md5'
+    datadir_base = os.path.expanduser(cache_dir)
+    if not os.access(datadir_base, os.W_OK):
+        datadir_base = os.path.join('/tmp', '.matchzoo')
+    datadir = os.path.join(datadir_base, cache_subdir)
+    if not os.path.exists(datadir):
+        os.makedirs(datadir)
+
+    if untar:
+        untar_fpath = os.path.join(datadir, fname)
+        fpath = untar_fpath + '.tar.gz'
+    else:
+        fpath = os.path.join(datadir, fname)
+
+    download = False
+    if os.path.exists(fpath):
+        if file_hash is not None:
+            if not validate_file(fpath, file_hash, algorithm=hash_algorithm):
+                print('A local file was found, but it seems to be '
+                      'incomplete or outdated because the file hash '
+                      'does not match the original value of file_hash.'
+                      ' We will re-download the data.')
+                download = True
+    else:
+        download = True
+
+    if download:
+        print('Downloading data from', origin)
```

Comment:
Need a space?

Reply:
The ',' will print a space, so it doesn't need a space anymore.